### ISSUE TITLE
venv: Checking the interpreter version after a Mu update upgrades Python provides a false pass

### ISSUE THREAD

**Issue by carlosperate** [mu-editor/mu#1776]
Adding some notes from memory (I wrote this issue properly when I opened it, but I must have failed to submit the description correctly 🤦‍♂️, so editing it again 26 days after creation):
- The Mu's venv module checks that the Python version in the venv matches the Python version running Mu
    - If the version doesn't match it fails the venv verification and creates a new one
- When using the installers, the virtual environment created for the user has a `python` executable that is esentially a symlink (or similar, depending on OS) that points to the `python` executable in the Mu installation directory
- When the Mu installation is updated, the `python` executable is replaced by a newer version
    - So let's say the previous Mu had Python 3.7 and the updated version has 3.8 
- The old venv, created with Python 3.7, has a symlink that still points to the same file, which now is Python 3.8, so the version check passes.
    - So basically, when Mu checks the user venv Python version, it ends up comparing the same Python executable
    - In this example the user venv might have packages for Python 3.7 that don't work with 3.8, even though the venv initialises correctly and Mu passes its checks

**Comment by carlosperate** [comment]
🤦‍♂️ I'm pretty sure I started editing the issue description with detailed information about the problem, but I must have never saved.
I'll see if I can remember what this was.

**Comment by carlosperate** [comment]
A possible solution to this is to store the Python version in the venv.json file.
